### ISSUE TITLE
Octave w/o ">>": @result{} splitting, could use @group in heuristics

### ISSUE THREAD

**Issue by cbm755** [catch22/octave-doctest#11]
Similarly to how we do @example?  Or should I just bite the bullet and run one line at a time, checking the output as we go along...

**Comment by cbm755** [comment]
Ideally want this to work:
````
## @example
## @group
## x = infsup (1 + eps);
## intervaltotext (x)
##   @result{} [1.0000000000000002, 1.0000000000000003]
## y = nextout (x);  % ** THIS LINE PROBLEMATIC FOR CURRENT APPROACH **
## intervaltotext (y)
##   @result{} [1, 1.0000000000000005]
## z = infsup (1);
## intervaltotext (z)
##   @result{} [1]
## @end group
## @end example
````

**Comment by cbm755** [comment]
careful with @group: as was discussed on octave-maintainers maillist, someone might use that to paginate long output...

**Comment by cbm755** [comment]
Another option here is to use a special comment "@c end result" or something like that.

I also wondered about adding a `@prompt` texinfo command, but have no looked into feasibility.

**Comment by cbm755** [comment]
So I like marking my examples with `>>`.  Others do not.

But such a thing simplifies doctest parsing.  Perhaps we could encourage the (optional) use of `@prompt` in texinfo, for (1) disambiguation and (2) flexibility in html/pdf documentation.

Then one could put one of the following texinfo macro definitions in the texinfo:
````
## @macro prompt{}
## >> 
## @end macro
````
or
````
## @macro prompt{}
## @end macro
````

Then use
````
## @example
## @group
## @prompt{} a = 6
## @result{} a = 6
##
## @prompt{} b = 2*a;
## @prompt{} c = 3*b
## @result{} c = 36
## @end group
## @end example
````
(note w/o prompt, its hard for doctest to tell if the `b = 2*a` is output or input)

@oheim what do you think?

**Comment by oheim** [comment]
I like the `@prompt` macro, because it adds semantic markup to the code. I also dislike it, because it complicates things.

Wouldn't it be possible, to use the indentation as a hint for the beginning of the next input line?

The scanner starts at the first line, which obviously is a prompt line. It may start with an optional “>>”. Then search for a subsequent line that either starts with “>>” or `@result`, whichever comes first.

If that matching line starts with “>>” or no such line is found until the end of the example block, then everything in between is classified as output.

If that matching line starts with `@result`, then everything up to that line is input and output starts in the `@result` line. Now we determine the minimum text indentation X of the (possibly many) input lines. Any consecutive lines after the `@result` line, which have an indentation greater than X are classified as output.

Since multiple output lines must be aligned in the example anyway and this is not possible when every line starts in column 1, the example author would naturally write the tests in a way that can easily be parsed this way.

Of course you have to ignore empty lines (or lines that contain only whitespace) and have to consider the comment markers (##  / %%).

````
[a, b] = deal (1, 2)
@result{}
    a =  1
    b =  2
c = 3
@result{} c =  3
d = 4, e = 5
@result{}
    d =  4
    e =  5
````

**Comment by cbm755** [comment]
@oheim I like this idea.  In principle could also relax the requirement to have `@result{}`.

I think my `parse_texinfo` function may need some refactoring... not sure when I'll get to this.

I've committed an xfail test based on your example, in case anyone else wants to tackle this.

**Comment by oheim** [comment]
Implemented as part of #107
